instrumentation/opentelemetry-api/opentelemetry-api-1.40/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/v1_40/incubator/trace/TracerTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.opentelemetryapi.v1_40.incubator.trace;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.opentelemetry.api.incubator.trace.ExtendedSpanBuilder;
+import io.opentelemetry.api.incubator.trace.ExtendedTracer;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceId;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class TracerTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Test
+  void isEnabled() {
+    Tracer disabledTracer = testing.getOpenTelemetry().getTracer("disabled-tracer");
+    Tracer enabledTracer = testing.getOpenTelemetry().getTracer("enabled-tracer");
+    testEnabled(disabledTracer, false);
+    testEnabled(enabledTracer, true);
+  }
+
+  private static void testEnabled(Tracer tracer, boolean expected) {
+    assertThat(tracer).isInstanceOf(ExtendedTracer.class);
+    assertThat(((ExtendedTracer) tracer).isEnabled()).isEqualTo(expected);
+  }
+
+  @Test
+  void extendedSpanBuilder() {
+    Tracer tracer = testing.getOpenTelemetry().getTracer("test");
+    SpanBuilder spanBuilder = tracer.spanBuilder("test");
+    assertThat(spanBuilder).isInstanceOf(ExtendedSpanBuilder.class);
+
+    ExtendedSpanBuilder builder = (ExtendedSpanBuilder) spanBuilder;
+    {
+      Span span = builder.startAndCall(Span::current);
```

Comment:
why pass a span as `SpanCallable`?

Reply:
It doesn't pass a span, it captures the span that was created by `startAndCall`. If you have better ideas how to test this method you are welcome to propose an alternative test.